source/elements/oneDNN/source/example.cpp

```diff
@@ -62,30 +68,36 @@ struct eltwise_forward {
 int N, C, H, W;
 
 // User-owned DPC++ objects
-sycl::device dev; // Device
-sycl::context ctx; // Context
-sycl::queue queue; // Queue
+sycl::device dev {sycl::gpu_selector {}}; // Device
+sycl::context ctx {dev}; // Context
+sycl::queue queue {sycl::gpu_selector {}}; // Queue
 std::vector<sycl::event> dependencies; // Input events dependencies
-sycl::buffer<float, 1> buf_src {sycl::range<1> {N * C * H * W}}; // Source
-sycl::buffer<float, 1> buf_dst {sycl::range<1> {N * C * H * W}}; // Results
+// Source
+float *buf_src = static_cast<float *>(
+        sycl::malloc_device((N * C * H * W) * sizeof(float), dev, ctx));
+// Results
+float *buf_dst = static_cast<float *>(
+        sycl::malloc_device((N * C * H * W) * sizeof(float), dev, ctx));
 
 // Create an engine encapsulating users' DPC++ GPU device and context
-dnnl::engine engine {dnnl::engine::kind::gpu, dev, ctx};
+dnnl::engine engine = dnnl::sycl_interop::make_engine(dev, ctx);
 // Create a stream encapsulating users' DPC++ GPU queue
-dnnl::stream stream {engine, queue};
+dnnl::stream stream = dnnl::sycl_interop::make_stream(engine, queue);
 // Create memory objects that use buf_src and buf_dst as the underlying storage
-dnnl::memory mem_src {{{N, C, H, W}, dnnl::memory::data_type::f32,
-                              dnnl::memory::format_tag::nhwc},
-        engine, buf_src};
-dnnl::memory mem_dst {{{N, C, H, W}, dnnl::memory::data_type::f32,
-                              dnnl::memory::format_tag::nhwc},
-        engine, buf_dst};
+dnnl::memory mem_src = dnnl::sycl_interop::make_memory(
+        {{N, C, H, W}, dnnl::memory::data_type::f32,
+                dnnl::memory::format_tag::nhwc},
+        engine, dnnl::memory_kind::usm_device, buf_src);
```

Comment:
memory with usm could be created with regular ctor. I suggest simplify the code here.

Reply:
On one hand I would like to keep it to emphasize the flexibility of the API and make as "default" for the user. On the other hand, the simplified constructor indeed may work well as a decent default option for most users.

Will update.